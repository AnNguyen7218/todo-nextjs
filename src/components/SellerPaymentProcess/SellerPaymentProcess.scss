@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.container {
  margin-bottom: gutter(24);

  @include break(md) {
    margin-bottom: gutter(0);
    flex: 1;
    &:not(:first-child) {
      margin-left: gutter(24);
    }
  }

  @include break(xl) {
    &:not(:first-child) {
      margin-left: gutter(48);
    }
  }
}

.logo {
  width: 3.6rem;
  height: 3.6rem;
  color: $purple;
  margin-bottom: gutter(12);

  @include break(md) {
    margin-bottom: gutter(12);
    width: 3.6rem;
    height: 3.6rem;
  }

  @include break(xl) {
    margin-bottom: gutter(16);
    width: 4.8rem;
    height: 4.8rem;
  }
}

.title {
  margin-bottom: gutter(4);
}
