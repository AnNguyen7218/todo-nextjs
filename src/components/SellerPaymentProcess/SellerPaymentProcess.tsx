import React from 'react';
import css from './SellerPaymentProcess.scss';

type Props = {
  process: SellerPaymentProcess;
};

const SellerPaymentProcess: React.FC<Props> = ({
  process: { icon: Icon, title, description },
}) => {
  return (
    <div className={css['container']}>
      <Icon className={css['logo']} />
      <h3 className={css['title']}>{title}</h3>
      <p>{description}</p>
    </div>
  );
};

export default SellerPaymentProcess;
