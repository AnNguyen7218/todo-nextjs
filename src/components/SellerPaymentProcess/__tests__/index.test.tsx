import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SellerPaymentProcess from '../SellerPaymentProcess';
import Config from 'assets/icons/ic-config.svg';

afterEach(cleanup);

const process = {
  icon: Config,
  title: 'Nec porta erat tinc',
  description:
    'At elit porta pellentesque aenean. Malesuada non aliquet mauris quisque. Congue enim pellentesque dictumst dictum.',
};

describe('SellerPaymentProcess Component', () => {
  test('SellerPaymentProcess should match snapshot', () => {
    const { container } = render(<SellerPaymentProcess process={process} />);
    expect(container).toMatchSnapshot();
  });
});
