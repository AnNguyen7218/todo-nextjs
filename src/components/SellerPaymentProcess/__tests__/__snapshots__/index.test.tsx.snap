// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerPaymentProcess Component SellerPaymentProcess should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <ic-config.svg
      classname="logo"
    />
    <h3
      class="title"
    >
      Nec porta erat tinc
    </h3>
    <p>
      At elit porta pellentesque aenean. Malesuada non aliquet mauris quisque. Congue enim pellentesque dictumst dictum.
    </p>
  </div>
</div>
`;
